internal/service/globalclusterconfig/resource_global_cluster_config.go

```diff
@@ -211,9 +212,38 @@ func resourceRead(ctx context.Context, d *schema.ResourceData, meta any) diag.Di
 }
 
 func resourceUpdate(ctx context.Context, d *schema.ResourceData, meta any) diag.Diagnostics {
-	return diag.Errorf("Updating a global cluster configuration resource is not allowed as it would " +
-		"leave the index and shard key on the related collection in an inconsistent state.\n" +
-		"Please read our official documentation for more information.")
+	connV2 := meta.(*config.MongoDBClient).AtlasV2
+	ids := conversion.DecodeStateID(d.Id())
+	projectID := ids["project_id"]
+	clusterName := ids["cluster_name"]
+
+	if d.HasChange("managed_namespaces") {
+		oldMN, newMN := d.GetChange("managed_namespaces")
+		oldList := oldMN.(*schema.Set).List()
+		newList := newMN.(*schema.Set).List()
+		if err := updateManagedNamespaces(ctx, connV2, projectID, clusterName, oldList, newList); err != nil {
+			return diag.FromErr(fmt.Errorf(errorGlobalClusterUpdate, clusterName, err))
+		}
+	}
+
+	if d.HasChange("custom_zone_mappings") {
+		oldZN, newZN := d.GetChange("custom_zone_mappings")
+		oldSet := oldZN.(*schema.Set)
+		newSet := newZN.(*schema.Set)
+		if err := updateCustomZoneMappings(ctx, connV2, projectID, clusterName, oldSet, newSet); err != nil {
+			return diag.FromErr(fmt.Errorf(errorGlobalClusterUpdate, clusterName, err))
+		}
+	}
+	return resourceRead(ctx, d, meta)
+}
+
+// Helper function to convert []map[string]any into []any
```

Comment:
```suggestion
// convertInterfaceSlice converts []map[string]any into []any
```

nit: Go idiomatic is to start func docs with func name as in the other funcs in this PR